doc/modules/ensemble.rst

```diff
@@ -223,10 +226,40 @@ or out-of-bag samples. This can be enabled by setting ``oob_score=True``.
 
 .. note::
 
-    The size of the model with the default parameters is :math:`O( M * N * log (N) )`,
-    where :math:`M` is the number of trees and :math:`N` is the number of samples.
-    In order to reduce the size of the model, you can change these parameters:
-    ``min_samples_split``, ``max_leaf_nodes``, ``max_depth`` and ``min_samples_leaf``.
+    The size of the model with the default parameters is :math:`O( M * N * log
+    (N) )`, where :math:`M` is the number of trees and :math:`N` is the number
+    of samples. In order to reduce the size of the model, you can change these
+    parameters: ``min_samples_split``, ``max_leaf_nodes``, ``max_depth`` and
+    ``min_samples_leaf``.
+
+Learning from imbalanced classes dataset
+........................................
+
+In some datasets, the number of samples per classes might vary tremendously
+(e.g 100 samples for a "majority" class for a single sample in a "minority"
+class). Learning from these imbalanced dataset is challenging. The tree
+criteria (i.e. gini or entropy) are sensitive to class imbalanced and will
+naturally favor the classes with the most samples given during ``fit``.
+
+The :class:`RandomForestClassifier` and :class:`ExtraTreesClassifier` provides
+a parameter `class_weight` with the option `"balanced_bootstrap"` to alleviate
+the bias induce by the class imbalanced. This strategy will provide a balanced
+bootstrap sample (i.e. equal number of samples from each class) to each tree
+of the ensemble during the ``fit`` process as proposed in [CLB2004]_.
```

Comment:
What is described here seems to simply be stratification, is that correct? Or is the difference that we sample with replacement?

Reply:
we do with replacement in addition

Reply:
OK I should rephrase, the exact definition is:

```
Draw a bootstrap sample from the minority class. Randomly drawthe same number of cases, with replacement, from the majority class.
```